include/openthread/message.h

```diff
@@ -103,8 +103,8 @@ typedef enum otMessagePriority
  */
 typedef struct otMessageSettings
 {
-    bool              mLinkSecurityEnabled; ///< TRUE if the message should be secured at Layer 2.
-    otMessagePriority mPriority;            ///< The message priority level.
+    bool    mLinkSecurityEnabled; ///< TRUE if the message should be secured at Layer 2.
+    uint8_t mPriority;            ///< Priority level (MUST be a `OT_MESSAGE_PRIORITY_*` from `otMessagePriority`).
```

Comment:
The `otMessagePriority` defines only 3 values (high, normal, low) but in the core we have a fourth propriety `net` (which we don't want to expose in public header).

The change of `mPriority` to be ` uint8_t` is the main reason we can now safely have the `Message::Settings`
inherit from the `otMessageSettings` and avoid the copying.

Note that if we use the `otMessagePriority` type, most tool-chain will use `uint8_t` for it but we cannot be sure of this as the behavior is tool-chain dependent and not specified by the c++ spec.

Reply:
Got it. I saw only the line 107 uses the enum `otMessagePriority`. After we  change the `otMessagePriority` to `uint8_t` here. No one uses `otMessagePriority` anymore. Could we change the 'typedef enum otMessagePriority'  to `enum`?

Reply:
I noticed the same (that the `otMessagePriority` is no longer used in OT itself). However, there may be some example app layer (next layer) code that may be using this definition (e.g., if I recall correctly, we used the `otMessagePriority` type in some internal project (RTOS platform) code in the past when mapping from other data buffers to `otMessage`).  I don't think keeping the enum named as  `otMessagePriority` would add any overhead in the code, so keeping it can help with any existing app (next layer) code that may be using it.